x/leverage/types/token.go

```diff
@@ -210,3 +198,53 @@ func DefaultRegistry() []Token {
 		defaultUmeeToken(),
 	}
 }
+
+// Validate performs validation on an SpecialAssetPair type
+func (p SpecialAssetPair) Validate() error {
+	if err := validateBaseDenoms(p.Collateral, p.Borrow); err != nil {
+		return err
+	}
+
+	// Collateral weight is non-negative and less than 1, or is exactly negative one.
+	if !p.CollateralWeight.Equal(sdk.MustNewDecFromStr("-1")) {
```

Comment:
Let's add a helper function to validate the `CollateralWeight` and execute it here and in the `SpecialAssetSet.Validate()`. Less code and easier to maintain if the validation changes